client/rest/src/test/java/org/elasticsearch/client/RestClientMultipleHostsIntegTests.java

```diff
@@ -201,27 +202,28 @@ public void onFailure(Exception exception) {
      * test what happens after calling
      */
     public void testNodeSelector() throws IOException {
-        Request request = new Request("GET", "/200");
-        RequestOptions.Builder options = request.getOptions().toBuilder();
-        options.setNodeSelector(firstPositionNodeSelector());
-        request.setOptions(options);
-        int rounds = between(1, 10);
-        for (int i = 0; i < rounds; i++) {
-            /*
-             * Run the request more than once to verify that the
-             * NodeSelector overrides the round robin behavior.
-             */
-            if (stoppedFirstHost) {
-                try {
-                    restClient.performRequest(request);
-                    fail("expected to fail to connect");
-                } catch (ConnectException e) {
-                    // This is different in windows and linux but this matches both.
-                    assertThat(e.getMessage(), startsWith("Connection refused"));
+        try (RestClient restClient = buildRestClient(firstPositionNodeSelector())) {
+            Request request = new Request("GET", "/200");
+            RequestOptions.Builder options = request.getOptions().toBuilder();
```

Comment:
I don't think we need `RequestOptions.Builder` at all here.

Reply:
ops :)